src/package-reference.js

```diff
@@ -55,15 +55,15 @@ export default class PackageReference {
   permissions: {[key: string]: boolean};
   remote: PackageRemote;
   registry: RegistryNames;
-  location: ?string;
+  locations: Array<string>;
   resolver: PackageResolver;
 
   setFresh(fresh: boolean) {
     this.fresh = fresh;
   }
 
-  setLocation(loc: string): string {
-    return (this.location = loc);
+  addLocation(loc: string): string {
```

Comment:
a package can have more than one location. For example, 3 workspaces depend on foo@2.0.0, and 2 workspaces depend on foo@1.0.0. foo@1.0.0 will be installed under both workspaces to avoid the conflict with the hoisted v2.

This was causing a bug where .bin links pointing to foo@1.0.0 would all point to a single copy of it, rather than the one installed for that package.

I also think if foo@1.0.0 has a postinstall script, it would only have run on one instance, but I never verified that. It shouldn't do that either way now though.